### ISSUE TITLE
routeNeedsPermission() and routeNeedsRole() not functional Laravel 5.2

### ISSUE THREAD

**Issue by bkuhl** [Zizaco/entrust#543]
Using `"zizaco/entrust": "5.2.x-dev"` with Laravel 5.2 I'm running into an issue using the route filters.  It looks like they aren't yet designed for Laravel 5.2.  The error I get when `clear-compiled` is run is:

```
  [BadMethodCallException]       
  Method filter does not exist.
```

Doing some digging it appears to be a result of the use of `Entrust::routeNeedsPermission()` and `Entrust::routeNeedsRole()` when it hits [`filter()`](https://github.com/Zizaco/entrust/blob/master/src/Entrust/Entrust.php#L153).

**Comment by yahyazini** [comment]
+1

**Comment by rahmatawaludin** [comment]
Have anyone look into this issue?

**Comment by yahyazini** [comment]
@rahmatawaludin I think the issue is mainly cause because of the use of Route filters because those  have been deprecated in preference of middleware.

**Comment by yahyazini** [comment]
I fixed this issue by using "zizaco/entrust": "5.2.x-dev" with Laravel 5.2

**Comment by hotvic** [comment]
+1

**Comment by hotvic** [comment]
Any news ? I need this because the middleware has no way to redirect user...

**Comment by bkuhl** [comment]
I switched to [bouncer](https://github.com/JosephSilber/bouncer) as future support for this package appears to be questionable.

**Comment by sugengsuprayogi** [comment]
+1

**Comment by EssamWD** [comment]
+1

**Comment by diegoalmesp** [comment]
@bkuhl is bouncer a good alternative to this package? I'm looking for something fully working with 5.2

**Comment by bkuhl** [comment]
Yes, it's build with a L5-friendly approach, so I think we can expect consistent Laravel 5 support with Bouncer.  The author of Bouncer also submitted a PR to Laravel that included some great middleware.  I'm using it and it's working well.  It didn't take long for me to switch.

**Comment by w3hacker** [comment]
+1

**Comment by larsbo** [comment]
:+1:

**Comment by marwankhanfar** [comment]
any one solved this issue ?

**Comment by quiquegarcia** [comment]
More opinions about Bouncer?

**Comment by dpovshed** [comment]
My 2 cents: I also spotted that [Short syntax route filter](https://github.com/Zizaco/entrust/blob/5a086c717dcc4e7fc41d312c93cd2d60dc0a0771/README.md#short-syntax-route-filter) failed to proceed with L5.2.

However, by using full format like:
```
    Route::get('reports', [
        'uses' => 'ReportController@reportsList',
        'middleware' => ['role:manager'],
    ]);
```

I was able to get things to work. Since that, everything works just fine, including GUI package I am using - https://github.com/acoustep/entrust-gui .

Hope this helps!

**Comment by marwankhanfar** [comment]
@quiquegarcia its probably too late for me to use Bouncer as I've already setup views and screens.
@dpovshed that method worked for me too , I like the routeNeedsRole cause it has redirect url option and its more generic .

Thanks

**Comment by segadora** [comment]
+1

**Comment by nerijuso** [comment]
+1

**Comment by WisdomSky** [comment]
+1

**Comment by xitude** [comment]
+1

**Comment by rygos** [comment]
+1 Any News?

**Comment by xitude** [comment]
Im worried support for this project has stopped. I may move to [bouncer](https://github.com/JosephSilber/bouncer) since the project owners last commit to this repo was **April**.

**Comment by oLDu67** [comment]
+1

**Comment by xiaoxiaocong** [comment]
route filters are not going away entirely after Laravel 5.0 However, middleware is now the preferred way to achieve the same functionality. See http://laravel.com/docs/master/middleware for info about how to use it. Middleware can be implemented to behave like either "before" or "after" filters. And it can be applied to all routes (called "global middleware"), or assigned to specific routes (by adding "'middleware' => 'auth'" for example to your route definitions in your routes.php file.

ref: https://stackoverflow.com/questions/34973173/method-filter-does-not-exist-error-in-laravel

**Comment by davidvleung** [comment]
The Short Syntax section of the README needs to be removed or indicated as deprecated if there is no intention of either fixing the `filter` issue or if middleware is truly going to be the path forward.  @Zizaco hopefully you can shed some light on the forward path.

**Comment by simtrax** [comment]
+1

**Comment by itsmekps** [comment]
+1
Entrust::routeNeedsRole('admin/advanced*', 'owner');   --> not working

showing error:
Attribute [filter] does not exist.

using laravel 5.5

help pls @Zizaco

**Comment by kostyan-org** [comment]
```
ErrorException (E_DEPRECATED)
Non-static method Zizaco\Entrust\Entrust::routeNeedsRole() should not be called statically
```

Entrust::routeNeedsRole('admin/advanced*', 'owner');

using laravel 5.7
